### DISCUSSION TITLE
Feedback: put off by delays

### DISCUSSION THREAD

**Discussion by mitranim** [go-task/task#601] [General]
* Started playing with `task`. Particularly interested in watch mode, because `make` requires external tools for that.
* Watch mode seems to have _stupidly_ huge delays between saving a file and running a task.
* `task --help` has no mention of delay or debounce.
* Want everything to be instant, with no delays.
* Went back to `make` with `watchexec -d=0`.

Not asking for any changes, just leaving the feedback here. Might be useful.

**Comment by andreynering** [comment]
Hi @mitranim,

Other than delays, has watch working good enough for you?

I admit the watch mode is something that got less attention than deserved, because fewer users use it AFAIK. So it may be less polished than ideal at the moment. Both feedback and contributions are welcome!

Regarding the interval, if you're willing to recompile the tool for a test you could change this constant to a lower value to see if it works for you:

https://github.com/go-task/task/blob/7e0346d6eb04a54e19106611851e4533fe23815c/watch.go#L19

We could potentially add an option to the Taskfile or a CLI argument (or both) to allow users to change that.

**Comment by mitranim** [comment]
Sounds like your watch mode uses polling. I would prefer if the tool used native FS notification APIs. For example, `https://github.com/rjeczalik/notify` provides a cross-platform solution for recursive watching that avoids polling on most platforms. There may be other alternatives. The difference is in CPU and battery use.

**Reply by andreynering** [reply]
Yes, the library we currently use uses polling: https://github.com/radovskyb/watcher

Seems that it has been unmaintained since 2019, so maybe we should eventually decide to replace it, indeed.
